static/app/views/lastKnownRouteContextProvider.tsx

```diff
@@ -0,0 +1,29 @@
+import {createContext, useContext} from 'react';
+
+import getRouteStringFromRoutes from 'sentry/utils/getRouteStringFromRoutes';
+import usePrevious from 'sentry/utils/usePrevious';
+import {useRoutes} from 'sentry/utils/useRoutes';
+
+interface Props {
+  children: React.ReactNode;
+}
+
+export const LastKnownRouteContext = createContext<string>('');
+
+export function useLastKnownRoute() {
+  return useContext(LastKnownRouteContext);
+}
+
+// This provider tracks the last known route that the user has navigated to.
```

Comment:
This should be a jsdoc style comment. Otherwise the LSP won’t pick this up as a documentation comment